src/btree/bt_discard.c

```diff
@@ -31,15 +31,14 @@ __wt_ref_out(WT_SESSION_IMPL *session, WT_REF *ref)
      */
     WT_ASSERT(session, S2BT(session)->evict_ref != ref);
 
-#ifdef HAVE_DIAGNOSTIC
     /*
      * Make sure no other thread has a hazard pointer on the page we are about to discard. This is
      * complicated by the fact that readers publish their hazard pointer before re-checking the page
      * state, so our check can race with readers without indicating a real problem. If we find a
      * hazard pointer, wait for it to be cleared.
      */
-    WT_ASSERT(session, __wt_hazard_check_assert(session, ref, true));
-#endif
+    WT_ASSERT_OPTIONAL(session, __wt_hazard_check_assert(session, ref, true),
+      "Attempting to evict page with hazard pointers");
```

Comment:
These messages are printed when the assert fires? I feel like the way you phrase them isn't that helpful to the user. Generally your phrasing is "doing xyz" which is in present tense, I think past tense works better here. So "Attempted to free a page with active hazard pointers." Similarly for the other error messages.